Packages/Nakama/Runtime/UnityWebRequestAdapter.cs

```diff
@@ -121,9 +121,20 @@ private UnityWebRequestAdapter()
             else if (www.isHttpError)
             {
                 var decoded = www.downloadHandler.text.FromJson<Dictionary<string, object>>();
-                errback(decoded == null
-                    ? new ApiResponseException(www.downloadHandler.text)
-                    : new ApiResponseException(www.responseCode, decoded["error"].ToString(), (int) decoded["code"]));
+
+                ApiResponseException e;
+
+                if (decoded == null)
+                {
+                    e = new ApiResponseException(www.downloadHandler.text);

```

Comment:
You could move this initialization to line 125. Then check if decoded not equal to `null` and replace the object.

Reply:
@novabyte I don't mind doing that -- I was wondering if that is generally considered more readable? That approach would be less lines of code. On the other hand, the advantage of current approach is it's only getting assigned to once which creates one less object. (Obviously we are talking fine grains of sand here, but I was just curious.)

Reply:
I think in general we always optimize to minimize allocations but in this case its an exceptional code path (low active execution) and will always allocate one object but many eliminate one overwrite with the allocation so its all small impact. :)